### DISCUSSION TITLE
Bug in simulate()?

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#161] [General]
Hello,
Not sure where I report a bug...but I think I found one. Translated to the spider data:

data(spider)
Y <- spider$abund
X <- spider$x
X <- scale(X)
X1 <- data.frame(soil.dry = X[,"soil.dry"])
X1
MGLM <- gllvm(Y, 
              X = X1, 
              formula = ~ soil.dry, 
              family = "gaussian", num.lv = 0)
Ysim <- simulate(MGLM, nsim = 1000, conditional = FALSE)
Error in `colnames<-`(`*tmp*`, value = colnames(object$X)) : 
  attempt to set 'colnames' on an object with less than two dimensions

Only happens if the data frame with the covariates contains exactly 1 covariate. If I add a second dummy covariate then it is fine...:-).

Kind regards,

Alain

**Comment by BertvanderVeen** [comment]
Definitely a bug. Addressed with commit in development repo.
